src/os/ObjectStore.h

```diff
@@ -766,6 +767,36 @@ class ObjectStore {
     const ghobject_t &oid  ///< [in] object
     ) = 0;
 
+  /**
+   * Iterate object map with user-provided callable
+   *
+   * Warning!  The callable is executed under lock on filestore
+   * operations in c.  Do not use filestore methods on c while
+   * when iterating.  (Filling in a transaction is no problem).
+   *
+   * @return error code, zero on success
+   */
+  struct omap_iter_seek_t {
+    std::string seek_position;
+    enum {
+      LOWER_BOUND,
+      UPPER_BOUND
+    } seek_type = LOWER_BOUND;
+    static omap_iter_seek_t min_lower_bound() { return {}; }
+  };
+  enum class omap_iter_ret_t {
+    STOP,
+    NEXT
+  };
+  virtual int omap_iterate(
+    CollectionHandle &c,   ///< [in] collection
+    const ghobject_t &oid, ///< [in] object
+    omap_iter_seek_t start_from, ///< [in] where the iterator should point to at the beginning
```

Comment:
if it's [IN] - why not const?
but I don't think it's really an 'in'. Let's verify

Reply:
I don't see a point for `const` for non-reference, non-pointer input arguments at the interface level. As this doesn't affect callers, it should be seen as implementation's detail.

Reply:
You are correct, of course. My mistake.